Source/ModelGroups/Mesh_PrimitiveMode.cs

```diff
@@ -1,488 +1,356 @@
+            UseFigure(imageList, "Indices");
+            UseFigure(imageList, "Indices_Points");
 
-            Runtime.Image figureIndices = new Runtime.Image
-            {
-                Uri = imageList.Find(e => e.Contains("Indices"))
-            };
-            Runtime.Image figurePointsIndices = new Runtime.Image
-            {
-                Uri = imageList.Find(e => e.Contains("Indices_Points"))
-            };
-            //Runtime.Image normalTexture = new Runtime.Image
-            //{
-            //    Uri = textures.Find(e => e.Contains("Normal_Plane"))
-            //};
-            //usedTextures.Add(normalTexture);
-            usedFigures.Add(figureIndices);
-            usedFigures.Add(figurePointsIndices);
+            // There are no common properties in this model group that are reported in the readme.
 
-            List<Vector3> noIndicesPositionsTriangles = new List<Vector3>()
-            {
-                new Vector3(-0.5f,-0.5f, 0.0f),
-                new Vector3( 0.5f,-0.5f, 0.0f),
-                new Vector3( 0.5f, 0.5f, 0.0f),
-                new Vector3(-0.5f,-0.5f, 0.0f),
-                new Vector3( 0.5f, 0.5f, 0.0f),
-                new Vector3(-0.5f, 0.5f, 0.0f),
-            };
-            List<Vector3> noIndicesPositionsLines = new List<Vector3>()
+            Model CreateModel(Action<List<Property>, Runtime.MeshPrimitive> setProperties)
             {
-                new Vector3( 0.5f,-0.5f, 0.0f),
-                new Vector3(-0.5f,-0.5f, 0.0f),
-                new Vector3(-0.5f,-0.5f, 0.0f),
-                new Vector3(-0.5f, 0.5f, 0.0f),
-                new Vector3(-0.5f, 0.5f, 0.0f),
-                new Vector3( 0.5f, 0.5f, 0.0f),
-                new Vector3( 0.5f, 0.5f, 0.0f),
-                new Vector3( 0.5f,-0.5f, 0.0f),
-            };
-            List<Vector3> noIndicesPositionsLineloopFan = new List<Vector3>()
-            {
-                new Vector3( 0.5f,-0.5f, 0.0f),
-                new Vector3( 0.5f, 0.5f, 0.0f),
-                new Vector3(-0.5f, 0.5f, 0.0f),
-                new Vector3(-0.5f,-0.5f, 0.0f),
-            };
+                var properties = new List<Property>();
+                var meshPrimitive = MeshPrimitive.CreateSinglePlane();
 
-            // Make a list of verticies for points with 1024 values
-            List<Vector3> noIndicesPositionsPoints = new List<Vector3>();
-            List<Vector3> cornerPoints = new List<Vector3>()
-            {
-                new Vector3(0.5f, -0.5f, 0.0f),
-                new Vector3(-0.5f, -0.5f, 0.0f),
-                new Vector3(-0.5f, 0.5f, 0.0f),
-                new Vector3(0.5f, 0.5f, 0.0f),
-                new Vector3(0.5f, -0.5f, 0.0f)
-            };
-            for (int corner = 0; corner < 4; corner++)
-            {
-                for (int x = 256; x > 0; x--)
+                // Apply the common properties to the gltf. 
+                meshPrimitive.Indices = null;
```

Comment:
would it be better to add a parameter to CreateSinglePlane to not write these out?

Reply:
I think that would be great. It would also be nice to have the same option for TextureCoords.